packages/frontend/src/creation-form/components/oracles-configuration/index.tsx

```diff
@@ -34,36 +50,75 @@ export const OraclesConfiguration = ({
     onNext,
 }: OraclesConfigurationProps): ReactElement => {
     const [data, setData] = useState(
-        oraclesData.reduce((accumulator: OracleDataMap, data, i) => {
+        oraclesData.reduce((accumulator: AugmentedOracleDataMap, data, i) => {
             accumulator[templates[i].id] = data;
             return accumulator;
         }, {})
     );
     const [disabled, setDisabled] = useState(true);
 
     useEffect(() => {
-        setDisabled(Object.keys(data).length !== templates.length);
-    }, [data, templates.length]);
+        try {
+            assertInitializationBundleGetterPresent(data);
+            setDisabled(false);
+        } catch (error) {
+            setDisabled(true);
+        }
+    }, [data]);
 
-    const handleDone = useCallback(
-        (id: number) => (initializationData: string, value: BigNumber) => {
-            setData({ ...data, [id]: { data: initializationData, value } });
+    const handleChange = useCallback(
+        (
+            templateId: number,
+            state: Partial<unknown>,
+            initializationBundleGetter?: OracleInitializationBundleGetter
+        ) => {
+            setData((prevData) => {
+                console.log({ initializationBundleGetter });
```

Comment:
`console.log` to remove here.

Reply:
Removed in 0c75ce4caf7d6fd2bb25cc7e62ce9db9a39d9374